base/libgit2/libgit2.jl

```diff
@@ -437,7 +447,27 @@ function revcount(repo::GitRepo, fst::AbstractString, snd::AbstractString)
     return (fc-1, sc-1)
 end
 
-""" git merge [--ff-only] [<committish> | FETCH_HEAD] """
+"""
+    merge!(repo::GitRepo; committish::AbstractString="", branch::AbstractString="", fastforward::Bool=false, merge_opts::MergeOptions=MergeOptions(), checkout_opts::CheckoutOptions=CheckoutOptions()) -> Bool
+
+Perform a git merge on the repository `repo`, merging commits
+with diverging history into the current branch. Returns `true`
+if the merge succeeded, `false` if not.
+
+The keyword arguments are:
+  * Merge the named commit(s) in `committish`.
+  * Merge the branch `branch` and all its commits since it diverged from the current branch.
+  * If `fastforward` is `true`, only merge if the merge is a fast-forward (the current branch
+    head is an ancestor of the commits to be merged), otherwise refuse to merge and return
+    `false`. This is equivalent to the git CLI option `--ff-only`.
+  * `merge_opts` specifies options for the merge, such as merge strategy in case of conflicts.
+    For more information, see [`MergeOptions()`](@ref).
+  * `checkout_opts` specifies options for the checkout step. For more information, see
+    [`CheckoutOptions()`](@ref).
+
+Equivalent to `git merge [--ff-only] [<committish> | FETCH_HEAD]`, where `branch` plays the role
+of `FETCH_HEAD`.
```

Comment:
I did it for consistency with the git docs. Since ours are so barebones it's natural a user/API writer would look there to cross reference.